src/mapblock_mesh.cpp

```diff
@@ -219,37 +225,24 @@ static u16 getSmoothLightCombined(const v3s16 &p,
 		} else {
 			ambient_occlusion++;
 		}
-		return f;
+		return f.light_propagates;
 	};
 
-	if (node_solid) {
-		ambient_occlusion = 3;
-		bool corner_obstructed = true;
-		for (int i = 0; i < 2; ++i) {
-			if (add_node(i).light_propagates)
-				corner_obstructed = false;
-		}
-		add_node(2);
-		add_node(3);
-		if (corner_obstructed)
-			ambient_occlusion++;
-		else
-			add_node(4);
-	} else {
-		std::array<bool, 4> obstructed = {{ 1, 1, 1, 1 }};
-		add_node(0);
-		bool opaque1 = !add_node(1).light_propagates;
-		bool opaque2 = !add_node(2).light_propagates;
-		bool opaque3 = !add_node(3).light_propagates;
-		obstructed[0] = opaque1 && opaque2;
-		obstructed[1] = opaque1 && opaque3;
-		obstructed[2] = opaque2 && opaque3;
-		for (int k = 0; k < 4; ++k) {
-			if (obstructed[k])
-				ambient_occlusion++;
-			else if (add_node(k + 4).light_propagates)
-				obstructed[3] = false;
-		}
+	std::array<bool, 4> obstructed = {{ 1, 1, 1, 1 }};
```

Comment:
is this not better to use a bitset here ?

Reply:
bitset is size-optimized, not speed-optimized

Reply:
Seems optimum then, ok for me.